src/main/java/net/minecraftforge/common/property/ExtendedBlockState.java

```diff
@@ -133,13 +133,14 @@ protected ExtendedStateImplementation(Block block, ImmutableMap<IProperty<?>, Co
         }
 
         @Override
+        @SuppressWarnings("unchecked")
         public <V>V getValue(IUnlistedProperty<V> property)
         {
             if(!this.unlistedProperties.containsKey(property))
             {
                 throw new IllegalArgumentException("Cannot get unlisted property " + property + " as it does not exist in " + getBlock().getBlockState());
             }
-            return property.getType().cast(this.unlistedProperties.get(property).orElse(null));
+            return (V) this.unlistedProperties.get(property).orElse(null);
```

Comment:
This is not equivalent. Your code will cause a crash in whoever is calling this method, this cast is unchecked.